web/modules/custom/dbcdk_community/src/Form/QuarantineCreateForm.php

```diff
@@ -0,0 +1,177 @@
+use Drupal\Core\DependencyInjection\ContainerInjectionInterface;
+use Drupal\Core\Form\FormBase;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Url;
+
+/**
+ * Create a Community Quarantine.
+ */
+class QuarantineCreateForm extends FormBase implements ContainerInjectionInterface {
+
+  /**
+   * The date format the Community Client expects.
+   */
+  const DATE_FORMAT = 'Y-m-d';
+
+  /**
+   * The Community Service Profile object from the username context.
+   *
+   * @var \DBCDK\CommunityServices\Model\Profile $profile
+   */
+  protected $profile;
+
+  /**
+   * The DBCDK Community Service Quarantine API.
+   *
+   * @var \DBCDK\CommunityServices\Api\QuarantineApi $quarantineApi
+   */
+  protected $quarantineApi;
+
+  /**
+   * QuarantineCreateForm constructor.
+   *
+   * @param \DBCDK\CommunityServices\Model\Profile $profile
+   *   The Community Profile we wish to grant a quarantine.
+   * @param \DBCDK\CommunityServices\Api\QuarantineApi $quarantine_api
+   *   The Community Service Quarantine API.
+   */
+  public function __construct(Profile $profile, QuarantineApi $quarantine_api) {
+    $this->profile = $profile;
+    $this->quarantineApi = $quarantine_api;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container) {
+    try {
+      $profile_filter = [
+        'where' => [
+          'username' => $container->get('request_stack')->getCurrentRequest()->get('username'),
+        ],
+      ];
+      /* @var \DBCDK\CommunityServices\Api\ProfileApi */
+      $profile_api = $container->get('dbcdk_community.api.profile');
+      $profile = $profile_api->profileFindOne(json_encode($profile_filter));
+    }
+    catch (ApiException $e) {
+      \Drupal::logger('DBCDK Community Service')->error($e);
+      // Create an empty Profile instance. The constructor expects a Profile
+      // object and will cause a fatal error if it receives something else and
+      // we would like to display an empty page with a Drupal message in case of
+      // an error.
+      $profile = new Profile();
+    }
+
+    return new static(
+      $profile,
+      $container->get('dbcdk_community.api.quarantine')
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getFormId() {
+    return 'dbcdk_community_quarantine_create_form';
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function buildForm(array $form, FormStateInterface $form_state) {
+    // Defensive coding to make sure we only display the "create quarantine"
+    // form if we could find a Community Profile.
+    if (empty($this->profile->getUsername())) {
+      drupal_set_message($this->t('The system could not find the profile that should receive this quarantine. Please try again later or contact an administrator.'), 'error');
+      return $form;
+    }
+
+    $form['reason'] = [
+      '#type' => 'textarea',
+      '#title' => $this->t('Reason'),
+      '#required' => TRUE,
```

Comment:
I was unclear. I was thinking that it should be WYSIWYG.

As I understood it the reason field should be WYSIWYG as that allowed moderators to communicate with eathother about the background for the ban, include links to offending content etc.